src/components/LoginPopupup/LoginPopupup.js

```diff
@@ -0,0 +1,138 @@
+import PopupWithForm from "../PopupWithForm/PopupWithForm.js";
+import React from "react";
+import {useSelector} from "react-redux";
+import {authenticate, validateToken} from "../../utils/auth.js";
+import {txtErr} from "../../utils/constants.js";
+import {capitalizeFirstLetter} from "../../utils/utils.js";
+import mainApi from "../../utils/MainApi.js";
+import {saveToStorage} from "../../utils/StorageService.js";
+import {removeFromStorage} from "../../utils/StorageService.js";
+
+const LoginPopupup = ({
+  onChangeInput,
+  handlePopupMouseDown,
+  handlePopupToggleView,
+  isInputHaveKey,
+  setCurrentUser,
+  onFormSubmitted,
+  setIsLoggedIn
+}) => {
+  const {
+    type,
+    isOpen,
+    title,
+    isFormVaild,
+    btnSetting,
+    bottomLink,
+    serverError,
+    inputs
+  } = useSelector((state) => state.fromSettingModule);
+
+  const settingPopupWithForm = {
+    type,
+    isOpen,
+    title,
+    isFormVaild,
+    btnSetting,
+    bottomLink,
+    inputs
+  };
+  const currType = "login";
+
+  return (
+    <PopupWithForm
+      handlePopupMouseDown={handlePopupMouseDown}
+      handleSubmit={(e) => {
+        e.preventDefault();
+        onFormSubmitted({isDone: false});
+        authenticate({
```

Comment:
**Might be better**

it's better to make api requests only in `App` because `Login` and `Register` should not know how to do it. They should be independent from such logics